File: LektaAndroid/app/src/main/java/com/lekta/app/models/SubscriptionTier.kt
package com.lekta.app.models

enum class SubscriptionTier(
    val displayName: String,
    val maxProducts: Int,
    val maxDevices: Int,
    val badge: String
) {
    FREE("Gratuito", 250, 1, "🆓"),
    TIER1("Reportes mensuales", 250, 1, "📊"),
    TIER2("Reportes mensuales", 250, 1, "📊"),
    TIER3("Reportes mensuales", 250, 1, "📊"),
    FULL("Reportes mensuales", 250, 1, "📊");

    val includesCloudBackup: Boolean get() = false
    val includesReports: Boolean get() = this != FREE
    val lowStockAlerts: Boolean get() = false
    val dataExport: Boolean get() = includesReports
    val prioritySupport: Boolean get() = false

    val features: List<String>
        get() = buildList {
            add("Hasta $maxProducts productos")
            add("Sin abono mensual")
            if (includesReports) {
                add("Reportes mensuales de ventas")
                add("Exportar datos para administración")
            } else {
                add("Pagás comisión solo cuando cobrás")
            }
        }
}

File: WarehouseApp/Models/SubscriptionTier.swift
import Foundation

enum SubscriptionTier: String, Codable, CaseIterable {
    case free     = "free"
    case tier1    = "tier1"
    case tier2    = "tier2"
    case tier3    = "tier3"
    case full     = "full"

    var displayName: String {
        switch self {
        case .free:  return "Gratuito"
        case .tier1: return "Reportes mensuales"
        case .tier2: return "Reportes mensuales"
        case .tier3: return "Reportes mensuales"
        case .full:  return "Reportes mensuales"
        }
    }

    var maxProducts: Int {
        switch self {
        case .free:  return 250
        case .tier1: return 250
        case .tier2: return 250
        case .tier3: return 250
        case .full:  return 250
        }
    }

    var maxDevices: Int {
        1
    }

    var includesCloudBackup: Bool { false }
    var includesReports: Bool     { self != .free }
    var lowStockAlerts: Bool      { false }
    var dataExport: Bool          { includesReports }
    var prioritySupport: Bool     { false }

    var badge: String {
        switch self {
        case .free:  return "🆓"
        case .tier1: return "📊"
        case .tier2: return "📊"
        case .tier3: return "📊"
        case .full:  return "📊"
        }
    }

    var features: [String] {
        var f = ["Hasta \(maxProducts) productos", "Sin abono mensual"]
        if includesReports {
            f.append("Reportes mensuales de ventas")
            f.append("Exportar datos para administración")
        } else {
            f.append("Pagás comisión solo cuando cobrás")
        }
        return f
    }
}

extension SubscriptionTier: Comparable {
    static func < (lhs: SubscriptionTier, rhs: SubscriptionTier) -> Bool {
        let order: [SubscriptionTier] = [.free, .tier1, .tier2, .tier3, .full]
        return (order.firstIndex(of: lhs) ?? 0) < (order.firstIndex(of: rhs) ?? 0)
    }
}
